pkg/aws/bucket.go

```diff
@@ -42,22 +42,27 @@ import (
 
 type bucket struct {
 	*config.Bucket
-	storage *storage
+	account *accountProvider
 	s3      *s3.S3
 
+	storage *storage
+
 	access string
 	bucket *s3.Bucket
+	arn    string
 }
 
-func newBucket(bkt resource.Bucket, cfg *config.Bucket, s *s3.S3) (resource.ProviderBucket, error) {
+func newBucket(bkt resource.Bucket, cfg *config.Bucket, s *s3.S3, acc *accountProvider) (resource.ProviderBucket, error) {
 	log.Debug("Initializing AWS Bucket %q", cfg.Name())
 
 	b := &bucket{
 		Bucket:  cfg,
+		account: acc,
 		storage: bkt.Storage().ProviderStorage().(*storage),
 		s3:      s,
 	}
 	b.set(b.storage.bucketCache.find(b))
+	b.arn = "arn:aws:iam::" + acc.Provider.Data["number"] + ":role/" + b.Role()
```

Comment:
I would much prefer if we extended arn.go to support role arn's in addition to instance profile arns. I would like to get the arn string from the code rather than hardcoding it here.